app/pages/Packages/ImportPackage.tsx

```diff
@@ -1,170 +1,84 @@
-                }
-            )
-
-            task.setContents(await Promise.all(savedContents))
-
-            return savedTask
-        })
-    )
-}
-
-const saveContentWithRelations = async (content: Content): Promise<Content> => {
-    const savedContent = await saveAndGetModelOrFail(content)
-
-    const savedQuests = content.getQuests().map(async (quest) => {
-        quest.setContent(savedContent)
-        const savedQuest = await saveAndGetModelOrFail(quest)
-
-        savedQuest.setAnswers(
-            await Promise.all(
-                quest.getAnswers().map((answer) => {
-                    answer.setQuest(savedQuest)
-
-                    return saveAndGetModelOrFail(answer)
-                })
-            )
-        )
-
-        return savedQuest
-    })
-    savedContent.setQuests(await Promise.all(savedQuests))
-
-    return savedContent
-}
+import PackageForm, { PackageFormSubmitHandler } from "./PackageForm"
 
 const ImportPackage: React.FC<RouteComponentProps> = () => {
-    const { mutate: mutatePackages } = usePackagesContext()
     const { course } = useCourseContext()
+    const { reloadPackages } = usePackagesContext()
 
-    const createPackage = useCallback(
-        async function (title: string, slug: string) {
-            const values = { title, slug }
+    const [json, setJson] = useState(null)
 
-            const newPackage = new Package()
-            newPackage.patch(values)
-            newPackage.setCourse(course)
-
-            const savedPackage = await saveAndGetModelOrFail(newPackage)
-
-            await mutatePackages((packages) => {
-                return [...packages, savedPackage]
-            })
-
-            return savedPackage
-        },
-        [course, mutatePackages]
-    )
-
-    const onImport = useCallback<ChangeEventHandler<HTMLInputElement>>(
-        async function (e) {
-            if (e.target.files == null) {
-                return false
+    const onSubmit = useCallback<PackageFormSubmitHandler>(
+        async (values) => {
+            if (!json) {
+                return
             }
-
-            for (const file of e.target.files) {
-                const fileContent = new Promise<string>((resolve, reject) => {
-                    const reader = new FileReader()
-
-                    reader.onload = (event) => {
-                        resolve((event.target?.result || "") as string)
+            // create new Json variable because the linter wants me to
+            const newJson: any = json
+            newJson.package.title = values.title
+            newJson.package.slug = values.slug
+
+            // create API url
+            const url = new URL(window.location.href)
+            url.search = ""
+            url.hash = ""
+            url.pathname = "jsonapi.php/v1/packages/import/" + course.getApiId()
+
+            // create formData object and its content
+            const blob = new Blob([JSON.stringify(newJson)], {
```

Comment:
So könnt ihr das `newJson` Objekt mit any-Type vermeiden.

```suggestion
            const blob = new Blob([JSON.stringify({
                package: {
                    title: values.title,
                    slug: values.slug
                } 
            })], {
```

Reply:
Wir haben uns jetzt drauf geeinigt, hier doch einen any type zu nutzen (irgendwie übersichtlicher), haben jedoch `newJson` vor `const blob...` verschoben